cluster/src/main/java/org/apache/spark/streaming/JavaSnappyStreamingJob.java

```diff
@@ -18,28 +18,33 @@
 
 
 import com.typesafe.config.Config;
-import org.apache.spark.sql.JSparkJobValidation;
-import org.apache.spark.sql.JavaJobValidate;
-import org.apache.spark.sql.streaming.SnappyStreamingJob;
+import org.apache.spark.sql.SnappyJobValidate;
+import org.apache.spark.sql.SnappyJobValidation;
 import org.apache.spark.streaming.api.java.JavaSnappyStreamingContext;
+
+import spark.jobserver.SparkJobBase;
 import spark.jobserver.SparkJobValidation;
+import org.apache.spark.util.SnappyUtils;
+import org.apache.spark.util.Utils;
 
-public abstract class JavaSnappyStreamingJob implements SnappyStreamingJob {
+public abstract class JavaSnappyStreamingJob implements SparkJobBase {
 
-  abstract public  Object runJavaJob(JavaSnappyStreamingContext snc, Config jobConfig);
+  abstract public Object runSnappyJob(JavaSnappyStreamingContext snc, Config jobConfig);
```

Comment:
It was more to do with type resolution in case of scala generic type. The API proposed is looking cleaner if its running all the tests. As use only have to deal with two methods. Only input types will change.